feat(doc): more marketing (#731)
nojhan/liquidprompt: docs/overview.rst

- **Configurability**: All aspects of the Liquid Prompt are configurable, down to the core.

Features overview
=================

The Liquid Prompt have a good support of shell-related features,
various version control systems, and several software configuration environments.

To browse the complete list of capabilities, see the :doc:`config`.

Shell Essentials
----------------

Those are some of the most popular features:

- **Current path**: displays where you are,
  highlight the important parts (current directory, version control repository up),
  smart path shortening.
- **Last command execution time**: if the last executed command took too long, display how many time.
- **Battery level**: show what remain in the battery if it's close enough to be empty.
- **Username**: display the user name if it's not the login one.
- **Hostname**: indicate the hostname if over a remote connection, with host-specific color.
- **Exit code**: display the last command's exit code if it was an error.
- **Jobs**: show counters for background jobs, sleeping and detached ones.
- **Time**: display the current time, using clock icons if you want a compact view.

> - **Memory**: warn if you are short on memory.

Author: I don't know what I was thinking… :man_facepalming:

- **Filesystem permission**: indicate if the current directory is writable or not.
- **Remote shell**: discretly denote if you are connected through SSH, under Tmux and with XForwarding capability.
- **Shell level**: displays the number of nested shells if it starts being too much.
- **Proxy**: indicate whether a proxy is configured.
- **Temperature**: warn if the temperature goes too high.
- **Prompt switch**: command allowing you to rapidely go back to the default prompt, or the even most simple possible.

> - **Wifi**: shows how many bars of wireless network you can hope for.

Author: It was kind of the point, actually.

  if the user has sudo rights without triggering a security alert
  that will annoy the sysadmin.

Competitors
===========

All prompt systems out there tend to focus on some feature sets.
If you don't like the Liquid Prompt's design, you may be interested in one of those popular prompts:
- [Starship](https://starship.rs/): focused on showing the development context. Probably the largest support of languages detection.
- [Spaceship](https://spaceship-prompt.sh/): same spirit than Starship (withh a little bit less features), but only for Zsh.
- [Oh-My-Posh](https://ohmyposh.dev/): tries to support a little bit of everything, and thus does not really shine at something.
- [Powerlevel10k](https://github.com/romkatv/powerlevel10k): the most popular on Github, focused on fancy theming, but to the expense of the UX.
- [Powerline](https://github.com/powerline/powerline): primarily a daemon-based status line generator, hence with good support for latency-expensive features.
  Probably the largest feature set for various services status.
- [Pure](https://github.com/sindresorhus/pure): quite popular, despite focusing on being minimalistic (only supports Zsh, Git and a few shell features).

> The following table compares those prompts systems in details.

Author: I have revamped the README and moved this stuff in the doc.

- `Oh-My-Posh <https://ohmyposh.dev/>`_: tries to support a little bit of everything, and thus does not really shine at something. Only for Zsh.
- `Powerlevel10k <https://github.com/romkatv/powerlevel10k>`_: the most popular on Github, focused on fancy theming, but to the expense of the UX.
- `Powerline <https://github.com/powerline/powerline>`_: primarily a daemon-based status line generator, hence with good support for latency-expensive features. Probably the largest feature set for various services status.
- `Pure <https://github.com/sindresorhus/pure>`_: quite popular, despite focusing on being minimalist (only supports Zsh, Git, and a few shell features).

The following table compares those prompts systems in details.

In that table, the numbers in cells figure the level of quality of the feature.
*Popularity* is the sum of levels in the row.
*Support* lines are the sum of levels in the column, for each *category* section.
*Category* sections are sorted from top to bottom based on their average popularity.
Projects are sorted from left to right, based on their *support* score in the *essentials* section.

.. image:: prompts_comparison.svg

>    :alt: A long table with 10 columns and 155 lines.

Reviewer: Still needs some sort of key or explanation. What do the numbers mean? What is the "Popularity" column?

Also, all cells with text in them need a solid background, otherwise when viewing in a dark theme the text is unreadable.

Author: It is already explained on the paragraph before (I can put it as a note to make it clearer?).

Reviewer: I must be blind, sorry.

> the numbers in cells figure the level of quality of the feature.

That seems quite arbitrary and subjective. At the very least we should document some guidelines for how we came up with those numbers, so we can try to be consistent in the future, and so anyone wanting more details on the chart can understand why we reached the numbers we did.

Reviewer: And I still think there should be less transparency, this is what it looks like for me:
![image](https://user-images.githubusercontent.com/8375571/186209451-bf5878a9-e34c-40f7-9dbf-d416c6fcf979.png)

License
-------

Liquid Prompt is distributed under the [GNU Affero General Public License

> version 3](LICENSE).

Reviewer: This link syntax is not valid RST.

Liquid Prompt gives you a **carefully designed prompt** with useful information.
It shows you what you need when you need it.
You will notice what changes *when* it changes, saving time and frustration.
You can even use it with your favorite shell – Bash or zsh.

Below are screenshots of typical states that you would see in everyday use.

Using the default theme:

.. image:: theme/default-med.png
   :alt: 1& [user@server:~/liquidprompt] main ±

Using the Powerline theme:

.. image:: theme/included/powerline-med.png
   :alt:   server  user  ~  liquidprompt  1   main  

> Why Liquidprompt?

Reviewer: ```suggestion
Why Liquid Prompt?
```

Using the default theme:

.. image:: theme/default-med.png
   :alt: 1& [user@server:~/liquidprompt] main ±

Using the Powerline theme:

.. image:: theme/included/powerline-med.png
   :alt:   server  user  ~  liquidprompt  1   main  

There are many prompts configurations out there, but here is what makes Liquid Prompt stand out:

- **UX Design**: Liquid Prompt was very carefully design from the beginning, to allow for the best user experience.
  That is, it displays *meaningful* information with *minimal visual clutter* and *maximum readability*.
  While most of the other prompts are focused on aligning as much colored "segments" as possible on top of each others to look fancy,
  Liquid Prompt focus on what really matters.
- **Ease of use**: Liquid Prompt is written in (*heavily optimized*) portable Shell, and its installation is as easy as copying files.
  It also provides pre-defined themes, enabled by a simple command in your shell configuration.
  Configuration is as simple as editing text files with key/value pairs. No complex format to learn.
- **Shell features first**: Liquid Prompt helps you being aware of all the powerfull features a modern shell can provide.

>   While most of the other prompts focus on covering a lot of (boring) software version tags,

Reviewer: ```suggestion
  While most other prompts focus on covering a lot of (boring) software version tags,
```

  Liquid Prompt shows you the features that are actually useful while you actually use the shell.
- **Configurability**: All aspects of Liquid Prompt are configurable, down to the core.

Features overview
=================

> Liquid Prompt have a good support of shell-related features,

Reviewer: ```suggestion
Liquid Prompt has good support of shell-related features,
```

various version control systems, and several software configuration environments.

> To browse the complete list of capabilities, see the :doc:`config`.

Reviewer: ```suggestion
To browse the complete list of capabilities, see :doc:`config`.
```

Shell Essentials
----------------

> Those are some of the most popular features:

Reviewer: ```suggestion
These are some of the most popular features:
```

- **Current path**: displays where you are,

>   highlight the important parts (current directory, version control repository up),

Reviewer: ```suggestion
  highlighting the important parts (current directory, version control repository root),
```

  smart path shortening.
- **Last command execution time**: if the last executed command took too long, display how much time it took.

> - **Battery level**: show what charge remains in the battery if it's close enough to be empty.

Reviewer: ```suggestion
- **Battery level**: show what charge remains in the battery if it's close enough to empty.
```

- **Username**: display the user name if it's not the login one.
- **Hostname**: indicate the hostname if over a remote connection, with host-specific color.
- **Exit code**: display the last command's exit code if it was an error.
- **Jobs**: show counters for background, sleeping, and detached jobs.
- **Time**: display the current time, using clock icons if you want a compact view.
- **Filesystem permission**: indicate if the current directory is writable or not.
- **Remote shell**: discretly denote if you are connected through SSH, under Tmux and with XForwarding capability.
- **Shell level**: displays the number of nested shells if it starts being too much.
- **Proxy**: indicate whether a proxy is configured.
- **Temperature**: warn if the temperature goes too high.

> - **Prompt switch**: command allowing you to rapidely go back to the default prompt, or the even most simple possible.

Reviewer: ```suggestion
- **Prompt switch**: command allowing you to rapidly go back to the default prompt, or the even most simple possible.
```

  the display of Liquid Prompt (use :attr:`LP_DISABLED_VCS_PATHS` to avoid that).
- Subversion repositories cannot display commits to be pushed because
  that's not how Subversion works
- The window's title escape sequence may not work properly on some terminals
  (like `xterm-256`).
- The analog clock requires a Unicode-aware terminal and at least a
  sufficiently complete font on your system. The `Symbola <https://dn-works.com/ufas/>`_
  font, designed by Georges Douros, is known to work well. On Debian or Ubuntu
  install try the `fonts-symbola` or `ttf-ancient-fonts` package.
  "Powerline" and "Nerd" fonts also have the approriate symbols.
- The "sudo" feature is disabled by default as there is no way to detect
  if the user has sudo rights without triggering a security alert
  that will annoy the sysadmin.

Competitors
-----------

All prompt systems tend to focus on some feature sets.
If you don't like Liquid Prompt's design, you may be interested in one of those popular prompts:

- `Starship <https://starship.rs/>`_: focused on showing development contexts. Probably the largest set of detected languages.

> - `Spaceship <https://spaceship-prompt.sh/>`_: similar spirit to Starship (with a little bit fewer features), but only for Zsh.

Reviewer: ```suggestion
- `Spaceship <https://spaceship-prompt.sh/>`_: similar spirit to Starship (with a few fewer features), but only for Zsh.
```

Author: One learn everyday, I would have bet this wasn't correct :laughing:

> - `Oh-My-Posh <https://ohmyposh.dev/>`_: tries to support a little bit of everything, and thus does not really shine at something. Only for Zsh.

Reviewer: ```suggestion
- `Oh-My-Posh <https://ohmyposh.dev/>`_: tries to support a little bit of everything, and thus does not really shine at anything. Only for Zsh.
```

> - `Powerlevel10k <https://github.com/romkatv/powerlevel10k>`_: the most popular on Github, focused on fancy theming, but to the expense of the UX.

Reviewer: ```suggestion
- `Powerlevel10k <https://github.com/romkatv/powerlevel10k>`_: the most popular on Github, focused on fancy theming, but to the expense of the UX. Only supports Zsh.
```

Author: Sorry, was lost in the rebase mayhem :disappointed:

- `Powerline <https://github.com/powerline/powerline>`_: primarily a daemon-based status line generator, hence with good support for latency-expensive features. Probably the largest feature set for various services status.
- `Pure <https://github.com/sindresorhus/pure>`_: quite popular, despite focusing on being minimalist (only supports Zsh, Git, and a few shell features).

The following table compares those prompts systems in details.

.. warning:: Those information have been gathered by *nojhan*, one of the author of Liquid Prompt.
          As such, they are highly subjective. Judgements made about the levels of support are
          extremely arbitrary. Take this with a grain of salt.

> In that table, the numbers in cells figure the level of quality of the feature.

Reviewer: ```suggestion
In this table, the numbers in cells figure the level of quality of the feature.

```
Needs the newline to enable bullet points below.
